editor/src/components/canvas/controls/grid-controls-ruler-markers.tsx

```diff
@@ -1,82 +1,116 @@
 import React from 'react'
-import { colorTheme } from '../../../uuiui'
+import type { UtopiColor } from '../../../uuiui'
 
-export type RulerMarkerType = 'span-start' | 'span-end' | 'auto' | 'pinned'
+export type RulerMarkerType = 'span-start' | 'span-end' | 'auto' | 'auto-short' | 'pinned'
 
-const upFacingTriangle = (
-  <svg width='11' height='11' xmlns='http://www.w3.org/2000/svg'>
-    <polygon
-      points='5,1 10,10 0,10'
-      fill={colorTheme.primary.value}
-      stroke={colorTheme.white.value}
-      strokeWidth='0.5'
-    />
-  </svg>
-)
+function upFacingTriangle(fillColor: UtopiColor, scale: number): React.ReactNode {
```

Comment:
small nitpick - we can have a component:
```tsx
function MarkerSVG({ scale, children }: MarkerSVGProps) {
  return (
    <svg
      width={`${11 / scale}`}
      height={`${11 / scale}`}
      viewBox='0 0 11 11'
      xmlns='http://www.w3.org/2000/svg'
    >
      {children}
    </svg>
  )
}
```
to be reused instead of all of the svgs:
```tsx
function upFacingTriangle(fillColor: UtopiColor, scale: number): React.ReactNode {
  return (
    <MarkerSVG scale={scale}>
      <polygon points='5.5,0 11,11 0,11' fill={fillColor.value} />
    </MarkerSVG>
  )
}

function rightFacingTriangle(fillColor: UtopiColor, scale: number): React.ReactNode {
  return (
    <MarkerSVG scale={scale}>
      <polygon points='11,5.5 0,0 0,11' fill={fillColor.value} />
    </MarkerSVG>
  )
}

function downFacingTriangle(fillColor: UtopiColor, scale: number): React.ReactNode {
  return (
    <MarkerSVG scale={scale}>
      <polygon points='5.5,11 0,0 11,0' fill={fillColor.value} />
    </MarkerSVG>
  )
}

function leftFacingTriangle(fillColor: UtopiColor, scale: number): React.ReactNode {
  return (
    <MarkerSVG scale={scale}>
      <polygon points='0,5.5 11,0 11,11' fill={fillColor.value} />
    </MarkerSVG>
  )
}

function regularVerticalPipe(fillColor: UtopiColor, scale: number): React.ReactNode {
  return (
    <MarkerSVG scale={scale}>
      <rect x='4' y='0' width='3' height='11' fill={fillColor.value} />
    </MarkerSVG>
  )
}

function regularHorizontalPipe(fillColor: UtopiColor, scale: number): React.ReactNode {
  return (
    <MarkerSVG scale={scale}>
      <rect x='0' y='4' width='11' height='3' fill={fillColor.value} />
    </MarkerSVG>
  )
}

function shortVerticalPipe(fillColor: UtopiColor, scale: number): React.ReactNode {
  return (
    <MarkerSVG scale={scale}>
      <rect x='4' y='2' width='3' height='9' fill={fillColor.value} />
    </MarkerSVG>
  )
}

function shortHorizontalPipe(fillColor: UtopiColor, scale: number): React.ReactNode {
  return (
    <MarkerSVG scale={scale}>
      <rect x='2' y='4' width='9' height='3' fill={fillColor.value} />
    </MarkerSVG>
  )
}
```

(but that's definitely just an optimization)

Reply:
Fixed in https://github.com/concrete-utopia/utopia/pull/6678/commits/6425c8b185eac8790ce7e6277e16d2380609ccb2.